### PR TITLE
api: Update swagger.yaml for configs

### PR DESCRIPTION
**Pull request by aaronlehmann**
This adds the config endpoints to swagger.yaml. Extracted from #33702.

ping @thaJeztah

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -2176,6 +2176,37 @@ definitions:
                     SecretName is the name of the secret that this references, but this is just provided for
                     lookup/display purposes. The secret in the reference will be identified by its ID.
                   type: "string"
+          Configs:
+            description: "Configs contains references to zero or more configs that will be exposed to the service."
+            type: "array"
+            items:
+              type: "object"
+              properties:
```
Comment (thaJeztah) [inline api/swagger.yaml]: Guess we should extract this as `ConfigReference` + `ConfigReferenceFileTarget ` definitions as well (https://github.com/moby/moby/blob/69c35dad8e7ec21de32d42b9dd606d3416ae1566/api/types/swarm/config.go#L18-L31)

Not needed for this PR, I'll have a look at those in a follow-up 👍

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
needs a rebase now @aaronlehmann

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Comment by aaronlehmann** [comment]
Rebased

**Review by thaJeztah — APPROVED** [review]
still LGTM

**Pull request merged**